qargparse.py

```diff
@@ -822,8 +832,8 @@ class Enum(QArgument):
     """
 
     def __init__(self, name, **kwargs):
-        kwargs["default"] = kwargs.pop("default", None)
-        kwargs["items"] = kwargs.get("items", [])
+        kwargs["items"] = kwargs.get("items", ["Default"])
+        kwargs["default"] = kwargs.pop("default", kwargs["items"][0])
```

Comment:
TypeError raised if `kwargs["items"]` is a  generator :/
```
TypeError: 'generator' object is not subscriptable
```
But shouldn't make it to `list` here as well, the current use case is to run the generator on `create`.
🤔

Reply:
Oh, change to `kwargs.pop("default", 0)` should work ? since looks like `Enum` is now all index-based ?

Reply:
Can confirm changing the line
```python
kwargs["default"] = kwargs.pop("default", kwargs["items"][0])
```
into
```python
kwargs["default"] = kwargs.pop("default", 0)
```
works. :D

Reply:
Nice, this should work nicely with it also being able to take an integer. Because otherwise, we would have needed to fetch the first string-item from that "items" list.